### PR TITLE
Move some WebKit code to WebKit framework

### PR DESCRIPTION
**Pull request by pvollan**
#### 676f48d6af5f4b9101395ad4ee76f64d23317e18
<pre>
Move some WebKit code to WebKit framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=265324">https://bugs.webkit.org/show_bug.cgi?id=265324</a>
<a href="https://rdar.apple.com/118776213">rdar://118776213</a>

Reviewed by Brent Fulgham.

Move some WebKit code related to WebKit process extensions to WebKit framework. This change enables
us to use new assertion API to take out an assertion on the Networking process when holding locked
files, which is also included in this patch.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::setIsHoldingLockedFiles):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::aqcuireLockedFileGrant):
(WebKit::NetworkProcess::invalidateGrant):
(WebKit::NetworkProcess::hasAcquiredGrant const):
* Source/WebKit/Shared/AuxiliaryProcessExtensions/AuxiliaryProcessExtensionBridge.h:
* Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift:
(Grant.invalidate):
(NetworkingProcessExtension.grant(_:name:)):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/271478@main">https://commits.webkit.org/271478@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7fc0483e96880b312405d5bed97b5f002a5b1f1c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28539 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7183 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29926 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31067 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25991 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9292 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4552 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28809 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5938 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24553 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5187 "Found 1 new API test failure: /WPE/TestWebKitWebView:/webkit/WebKitWebView/notification (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/5314 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31780 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26138 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/25995 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31582 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5278 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3447 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29357 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6881 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6827 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5737 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5797 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/Configurations/BaseExtension.xcconfig
Comment (brentfulgham) [inline Source/WebKit/Configurations/BaseExtension.xcconfig:37]: This pattern of hard-coding specific iOS releases seems unusual. I'd prefer for @emw-apple to review these bits.

**Review by brentfulgham — APPROVED** [review]
The code changes look good to me, but I'm less confident about the hard-coded iPhone OS release numbers. I'm approving, but would appreciate it if you could check with @emw-apple before landing.

File: Source/WebKit/Configurations/BaseExtension.xcconfig
Comment (emw-apple) [reply Source/WebKit/Configurations/BaseExtension.xcconfig:37]: This was my request, actually. We need to enable this functionality for all builds _after_ iOS 17.3. The more normal thing to do is to use build setting macros from `WebKitTargetConditionals.xcconfig`, but we generally only define those for major OS releases.

File: Source/WebKit/Shared/Cocoa/WKProcessExtension.mm
Comment (emw-apple) [inline Source/WebKit/Shared/Cocoa/WKProcessExtension.mm]: nit: This and the other NSLog calls in this file feel like debugging statements to me, that probably shouldn't make it to production and will only cause log spam. What do you think?

File: Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift
Comment (emw-apple) [inline Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift:71]: Instead of having an abstract `-[WKProcessExtension grant:name:]` method that you override from Swift, you could declare this extension method as `@objc`. Then, from `NetworkProcessCocoa.mm`, you could do a `respondsToSelector(@selector(grant:name:))` to check that the method exists (assertion-failing if it does not), then call it.

That might be easier to follow than the abstract method, which should never be called, but has to be written nonetheless.

File: Source/WebKit/Shared/Cocoa/WKProcessExtension.mm
Comment (pvollan) [reply Source/WebKit/Shared/Cocoa/WKProcessExtension.mm]: Fixed.

Thanks for reviewing!

File: Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift
Comment (pvollan) [reply Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift:71]: That is a good point, although I think it also would be good to avoid the respondsToSelector call. I went ahead and landed as-is, but please let me know if you still think this should be changed, and I can follow up in a new patch :)

**Pull request merged**
